### DISCUSSION TITLE
Collection size limitation to 8000 byte

### DISCUSSION THREAD

**Discussion by gorgicze** [litedb-org/LiteDB#2089] [Q&A]
The LiteDB is limited to use 8000 byte for the collection name and also used for the HeaderPage buffer size.
Can you explain, why was limited for this value? Why was decided to have this limit?

**Comment by ioagr** [comment]
Page size is limited to 8192 bytes for performance reasons. I also tried doubling the page size, but the Insert() and Find() functions became very slow, so I gave up.
HeaderPage is the same size as the page size (192 used for pragma and 8000 for collection names). The reason for this, I think, is because it is easier to read fixed size blocks from HDD than to calculate the position of the pages according to the HeaderPage size. But LiteDB developers can tell more details.
I would also like to be a defined constant from which I can change the size of the HeaderPage without changing the size of the rest of the Pages.
